Block/Adminhtml/Edit/Tab/Nodes.php

```diff
@@ -175,7 +191,43 @@ private function renderNodeList($level, $parent, $data)
         }
         $nodes = $data[$level][$parent];
         $menu = [];
+
+        // Create store view lookup array
+        $storeViewLabels = [];
+        $websites = $this->systemStore->getWebsiteCollection();
+        $websiteNames = [];
+
+        // Create website name lookup array
+        foreach ($websites as $website) {
+            $websiteNames[$website->getId()] = $website->getName();
+        }
+
+        foreach ($this->systemStore->getStoreCollection() as $store) {
+            if ($store->isActive()) {
+                $websiteName = isset($websiteNames[$store->getWebsiteId()])
+                    ? $websiteNames[$store->getWebsiteId()]
+                    : '';
+                $storeViewLabels[$store->getId()] = [
+                    'value' => $store->getId(),
+                    'label' => sprintf('%s -> %s', $websiteName, $store->getName())
+                ];
+            }
+        }
+
```

Comment:
It looks like duplicate piece of code. I think you can use the function getStoreViews().